Adapt contribution guides to fight AI slop (#19758)
openSUSE/open-build-service: CONTRIBUTING.md

If one of the goes wrong for your pull request please address the issue.

> ## AI Guidance
> 
> Using AI tools to help write your PR is acceptable, but as the author, you are responsible for understanding every change. If you used AI tools in preparing your PR, you must disclose this in the description of your PR. Listing AI tooling as a co-author, co-signing commits using an AI tool, or using the assisted-by, co-developed or similar commit trailer is not allowed.
> 
> Do not leave the first review of AI generated changes to the reviewers. Verify the changes (code review, testing, etc.) before submitting your PR. Reviewers may ask questions about your AI-assisted code, and if you cannot explain why a change was made, the PR will be closed.
> 
> When responding to review comments, you must do so without relying on AI tools. Reviewers want to engage directly with you, not with generated responses. If you do not engage directly with reviewers, the PR will be closed.
> 

Reviewer: ```suggestion
## AI Guidance

Using AI tools to help write your contribution is acceptable, as long as you follow the principles described below. 

As the author, you are *accountable* for the code you submit. Contributing to Open Build Service means vouching for the quality, license compliance, and utility of your contribution.

Do not leave the first review of AI generated changes to the reviewers. Verify the changes (code review, testing, etc.) before submitting your contribution. Reviewers may ask questions about your AI-assisted code, and if you cannot explain why a change was made, the PR will be closed.

When responding to review comments, you must do so without relying on AI tools. Reviewers want to engage directly with you, not with generated responses from your AI tool. If you do not engage directly with reviewers, the PR will be closed.

You **MUST** disclose if you have used generative AI tools in preparing significant portions of your contribution. The recommended method is an `Assisted-by:` [git commit message trailer](https://git-scm.com/docs/SubmittingPatches#commit-trailers) like

`Assisted-by: AGENT_NAME:MODEL_VERSION`

If that is not possible make the fact known in the in the description of your pull request in some other form.
```

Reviewer: I think we rather want to **require** `Assisted-by` commit trailer than forbid it.

Author: Alright

If one of the goes wrong for your pull request please address the issue.

## AI Guidance

Using AI tools to help write your PR is acceptable, but as the author, you are responsible for understanding every change. If you used AI tools in preparing your PR, you must disclose this in the description of your PR. Listing AI tooling as a co-author, co-signing commits using an AI tool, or using the assisted-by, co-developed or similar commit trailer is not allowed.

> All contributions must follow this contributions policies and use commit messages that align with the policy.

Reviewer: That is what this guidelines says, seems redundant to me to mention this again.

Author: Removed

> Large AI generated PRs and AI generated commit messages are not allowed.

Reviewer: I would leave this sentence out, we already say you are supposed to discuss large changes in advance and that PRs and commits should be small and targeted.

Author: Removed

If one of the goes wrong for your pull request please address the issue.

## AI Guidance

Using AI tools to help write your contribution is acceptable, as long as you follow the principles described below. 

As the author, you are *accountable* for the code you submit. Contributing to Open Build Service means vouching for the quality, license compliance, and utility of your contribution.

Do not leave the first review of AI generated changes to the reviewers. Verify the changes (code review, testing, etc.) before submitting your contribution. Reviewers may ask questions about your AI-assisted code, and if you cannot explain why a change was made, the PR will be closed.

When responding to review comments, you must do so without relying on AI tools. Reviewers want to engage directly with you, not with generated responses from your AI tool. If you do not engage directly with reviewers, the PR will be closed.

> You **MUST** disclose if you have used generative AI tools in preparing significant portions of your contribution. The recommended method is an `Assisted-by:` [git commit message trailer](https://git-scm.com/docs/SubmittingPatches#commit-trailers) like

Reviewer: ```suggestion
You **MUST** disclose if you have used generative AI tools in preparing portions of your contribution. The recommended method is an `Assisted-by:` [git commit message trailer](https://git-scm.com/docs/SubmittingPatches#commit-trailers) like
```

Reviewer: suggested by @darix to remove ambiguity what is significant and what is not.

`Assisted-by: AGENT_NAME:MODEL_VERSION`

> If that is not possible make the fact known in the in the description of your pull request in some other form.

Reviewer: ```suggestion

```

Reviewer: suggested by @hellcp-work to remove the ambiguity. There really isn't any reason you would not be able to do this...
